'ARGOCD_AGENT_IN_CLUSTER=true make setup-e2e' fails on redis secrets

**Describe the bug**
The `ARGOCD_AGENT_IN_CLUSTER=true` env var behaviour has regressed.

**To reproduce**:

Run via in standard E2E test setup:
```
ARGOCD_AGENT_IN_CLUSTER=true make setup-e2e

# (...)
Waiting for deployment "argocd-server" rollout to finish: 1 old replicas are pending termination...
Waiting for deployment "argocd-server" rollout to finish: 1 old replicas are pending termination...
deployment "argocd-server" successfully rolled out
deployment.apps/argocd-repo-server restarted
Waiting for deployment "argocd-repo-server" rollout to finish: 0 out of 1 new replicas have been updated...
Waiting for deployment "argocd-repo-server" rollout to finish: 1 old replicas are pending termination...
Waiting for deployment "argocd-repo-server" rollout to finish: 1 old replicas are pending termination...
deployment "argocd-repo-server" successfully rolled out
deployment.apps/argocd-agent-principal restarted
Waiting for deployment "argocd-agent-principal" rollout to finish: 0 out of 1 new replicas have been updated...
Waiting for deployment "argocd-agent-principal" rollout to finish: 0 out of 1 new replicas have been updated...
Waiting for deployment "argocd-agent-principal" rollout to finish: 1 old replicas are pending termination...
error: deployment "argocd-agent-principal" exceeded its progress deadline
make: *** [Makefile:84: setup-e2e] Error 1
```

If you switch to argocd namespace on control plane, you will notice that principal pods aren't running:
```		
jgw@fedora:~/workspace/argo-cd/argocd-agent$ k get pods
NAME                                      READY   STATUS              RESTARTS   AGE
argocd-agent-principal-579b5bf89-xn9lz    0/1     ContainerCreating   0          12m
argocd-agent-principal-6478d7c5d4-lq8qw   0/1     ContainerCreating   0          12m
argocd-dex-server-694c74cbb8-8wp97        1/1     Running             0          13m
argocd-redis-85b9d55dc4-fx5t2             1/1     Running             0          13m
argocd-repo-server-68965bcff7-cxdbk       1/1     Running             0          12m
argocd-server-5888cf5fc5-bvbqw            1/1     Running             0          12m
jgw@fedora:~/workspace/argo-cd/argocd-agent$ 
```

According to k8s events this is due to missing redis secrets (names are mangled by vcluster):
```		
jgw@fedora:~/workspace/argo-cd/argocd-agent$ kubectl get events | grep -i warning
58s         Warning   FailedMount         pod/argocd-agent-principal-579b5bf89-xn9lz     MountVolume.SetUp failed for volume "redis-tls-ca" : secret "argocd-redis-tls-x-argocd-x-vcluster-control-plane" not found
3m          Warning   FailedMount         pod/argocd-agent-principal-579b5bf89-xn9lz     MountVolume.SetUp failed for volume "redis-proxy-server-tls" : secret "argocd-redis-proxy-tls-x-argocd-x-vcluster-control-plane" not found
13m         Warning   FailedMount         pod/argocd-agent-principal-5bb8656d85-svst2    MountVolume.SetUp failed for volume "redis-tls-ca" : secret "argocd-redis-tls-x-argocd-x-vcluster-control-plane" not found
13m         Warning   FailedMount         pod/argocd-agent-principal-5bb8656d85-svst2    MountVolume.SetUp failed for volume "redis-proxy-server-tls" : secret "argocd-redis-proxy-tls-x-argocd-x-vcluster-control-plane" not found
107s        Warning   FailedMount         pod/argocd-agent-principal-6478d7c5d4-lq8qw    MountVolume.SetUp failed for volume "redis-proxy-server-tls" : secret "argocd-redis-proxy-tls-x-argocd-x-vcluster-control-plane" not found
3m49s       Warning   FailedMount         pod/argocd-agent-principal-6478d7c5d4-lq8qw    MountVolume.SetUp failed for volume "redis-tls-ca" : secret "argocd-redis-tls-x-argocd-x-vcluster-control-plane" not found
```

This is a regression caused by #664 e.g. commit `c65aed83f107c9307a1c9769b1abe9577dac7cb5`: When I checked out the commit BEFORE this one, running `ARGOCD_AGENT_IN_CLUSTER=true make setup-e2e` worked as expected.

--- a/install/kubernetes/principal/principal-deployment.yaml
+++ b/install/kubernetes/principal/principal-deployment.yaml
@@ -371,11 +371,11 @@ spec:
             path: tls.crt
           - key: tls.key
             path: tls.key
-          optional: false
+          optional: true
       - name: redis-tls-ca
         secret:
           secretName: argocd-redis-tls
           items:
           - key: ca.crt
             path: ca.crt
-          optional: false
+          optional: true

